### DISCUSSION TITLE
How to compile and use on RPi4 ?

### DISCUSSION THREAD

**Discussion by AndKe** [libvips/libvips#2237] [Q&A]
I downloaded https://github.com/libvips/libvips/releases/download/v8.10.6/vips-8.10.6.tar.gz
did ./configure , make , and sudo make install.

still vips -v  shows the old version 8.7.4
how do I actually switch to the compiled version?

**Comment by jcupitt** [comment]
Hi @AndKe,

`configure` will put vips into `/usr/local`, so you have to have that first on your PATH, or you need to remove the system one. 

Removing the system one is probably easiest, just `apt remove libvips` I think.

**Comment by AndKe** [comment]
Thank you.

**Comment by AndKe** [comment]
I am now trying to solve this:
(vips:26906): VIPS-WARNING **: 19:33:36.580: ignoring palette (no quantisation support)
(vips:26906): VIPS-WARNING **: 19:33:36.580: Invalid color type/bit depth combination in IHDR
(vips:26906): VIPS-WARNING **: 19:33:36.580: Invalid IHDR data
- are you aware of any precompiled vips package for rpi4 ?

**Reply by AndKe** [reply]
or..of course, libraries/dependencies required for the errors above to be resolved?  (./configure  does not seem to reveal/mention IHDR or quantisation)

**Comment by jcupitt** [comment]
You need libimagequant for low-bit rate PNG. Have a look at the summary table at the end of configure's output.

**Comment by AndKe** [comment]
I already have packeges libimagequant0 libimagequant-dev 
The whole problem is:
```
vips copy ../test/test2.svg x.png[bitdepth=4,filter=8,compression=9]
(vips:7504): VIPS-WARNING **: 20:04:42.305: ignoring palette (no quantisation support)
(vips:7504): VIPS-WARNING **: 20:04:42.305: Invalid color type/bit depth combination in IHDR
(vips:7504): VIPS-WARNING **: 20:04:42.305: Invalid IHDR data
vips2png: unable to write to target x.png
```

Can you please take a look on my ./configure output ? - maybe it can reveal some missing component:
https://gist.github.com/AndKe/a5bca48499adfe9c8e6242c3df534f7a

**Reply by jcupitt** [reply]
Looks OK. Perhaps you're picking up an old version, somehow?

**Reply by AndKe** [reply]
I don't think so.. I got it from https://github.com/libvips/libvips/releases/download/v8.10.6/vips-8.10.6.tar.gz
and the version is: 
```
vips -v copy ../test/test2.svg x.png[bitdepth=4,filter=8,compression=9]
vips-8.10.6-Tue Mar 23 20:52:58 UTC 2021

(vips:12471): VIPS-WARNING **: 20:47:50.025: ignoring palette (no quantisation support)
(vips:12471): VIPS-WARNING **: 20:47:50.026: Invalid color type/bit depth combination in IHDR
(vips:12471): VIPS-WARNING **: 20:47:50.026: Invalid IHDR data
vips2png: unable to write to target x.png
```
can it be that the vips is not the issue, but a library it's using is old?  (this is the current raspberry distro, not very old one, and it is updated  - I also uninstalled the distro-provided  libvips-tools and libvips42

**Reply by jcupitt** [reply]
You could try:

```
ldd $(which vips) | grep quant
```

I see:

```
$ ldd $(which vips) | grep quant
	libimagequant.so.0 => /lib/x86_64-linux-gnu/libimagequant.so.0 (0x00007f1316a0c000)
```

**Comment by AndKe** [comment]
I get no output (ldd works, but there is nothing that matches that grep)
Thanks to your example, I could find that library in :
```
$ find /usr/lib -name libimagequant*
/usr/lib/arm-linux-gnueabihf/libimagequant.a
/usr/lib/arm-linux-gnueabihf/libimagequant.so
/usr/lib/arm-linux-gnueabihf/libimagequant.so.0
```

and my library configuration points to:
```
cat /etc/ld.so.conf.d/libc.conf 
# libc default configuration
/usr/local/lib
 ```

so I added "/usr/lib" to that file
rebooting and recompiling now...

**Reply by jcupitt** [reply]
You probably only need to run `sudo ldconfig` to rebuild the cache.

**Reply by kmartinez** [reply]
just spotted this - I cloned the repo on my pi4 but that was a few months ago - perhaps something has changed in Pi-OS  - I'll try that tar ball later

**Reply by AndKe** [reply]
@kmartinez  I did had sucecss with it in the end. The major issue was that libc.conf did not include the path to where the library was installed after make install.

**Reply by kmartinez** [reply]
Ah yes - I stick to my decades-old solution of adding PATH things to my .bashrc for vips ;-) and as I tend to build from src I never have the debian package installed in case of conflicts.
By the way I ran our old benckmark on my Pi4 to see how slow it was compared to modern CPUs:
https://github.com/libvips/libvips/wiki/Benchmarks
much faster than the pi3 - and very fast per GHz but quite slow overall ;-)

**Comment by jcupitt** [comment]
I bought an r-pi4 and I've fixed git master libvips on it. It now builds cleanly and passes the test suite. It might be worth trying that.
